resources/asciidoctor/lib/copy_images/copier.rb

```diff
@@ -66,8 +55,34 @@ def find_source(block, uri)
         end
       end
 
-      # We'll skip images we can't find but we should log something about it so
-      # we can fix them.
+      log_missing(block, checked)
+      nil
+    end
+
+    ##
+    # The directory roots to check for the file to copy
+    def roots_to_check(block)
+      to_check = [block.document.base_dir]
+
+      resources = block.document.attr 'resources'
+      return to_check unless resources
+      return to_check if resources.empty?
+
+      begin
+        to_check + CSV.parse_line(resources)
+      rescue CSV::MalformedCSVError => error
+        logger.error message_with_context "Error loading [resources]: #{error}",
+            :source_location => block.source_location
+        to_check
+      end
+    end
```

Comment:
Instead of writing individual comments, I thought it be easier to visualize if I rewrote the method and put the two comments inline:

```ruby
    def roots_to_check(block)
      to_check = [block.document.base_dir]

      resources = block.document.attr 'resources'
      # I find it clearer to use `if` to check both scenarios in which `to_check` should be returned, instead of having an `if` followed by an `unless`
      return to_check if resources.nil? || resources.empty?

      to_check + CSV.parse_line(resources)
    # There is an implicit `begin` for a method, so you can just rescue an error here. It's not necessary, I just thought I would point out some fancy Ruby. And it makes the method two whole lines shorter ;)
    rescue CSV::MalformedCSVError => error
      logger.error message_with_context "Error loading [resources]: #{error}",
          :source_location => block.source_location
      to_check
    end
```

Reply:
Hmmmm! I like being explicit about which line can through the error, but in this case it is petty obvious which one can. I'll take it!